graphcanvas/tests/test_graph_container.py

```diff
@@ -0,0 +1,186 @@
+
+    def test_draw(self):
+        container = self.create_graph_container()
+        gc = GraphicsContext(tuple(container.bounds))
+        # draw the contents of the container
+        container.draw(gc)
+        # find the expected position of each component and test that it is
+        # drawn there
+        self.assert_components_drawn(container)
+
+    def test_draw_directed_arrow_direction(self):
+        d = {'a':['b'], 'b':[]}
+        g = graph_from_dict(d)
+        container = GraphContainer(graph=g)
+        for node in g.nodes():
+            GraphNodeComponent(container=container, value=node)
+        gc = GraphicsContext(tuple(container.bounds))
+
+        # Node a is to the left of node b
+        container._layout_needed = False
+        container.components[0].x = 0.0
+        container.components[1].x = 100.0
+        container.draw(gc)
+        self.assert_components_drawn(container)
+
+        # Node a is to the right of node b
+        container._layout_needed = False
+        container.components[0].x = 100.0
+        container.components[1].x = 0.0
+        container.draw(gc)
+        self.assert_components_drawn(container)
+
+    def test_draw_no_layout(self):
+        container = self.create_graph_container()
+        gc = GraphicsContext(tuple(container.bounds))
+        # set _layout_needed to False
+        container._layout_needed = False
+        # draw the contents of the container
+        container.draw(gc)
+        # test that all components are at (0, 0)
+        zero_zero_list = container.components_at(0, 0)
+        self.assertTrue(
+            all([(comp in container.components) for comp in zero_zero_list])
+        )
+
+    def test_draw_not_directed(self):
+        d = {'a':['b'], 'b':['c', 'd'], 'c':[], 'd':[]}
+        g = graph_from_dict(d)
+        g = g.to_undirected()
+        container = GraphContainer(graph=g)
+        for node in g.nodes():
+            GraphNodeComponent(container=container, value=node)
+        gc = GraphicsContext(tuple(container.bounds))
+        container.draw(gc)
+        self.assert_components_drawn(container)
+
+    def test_weighted(self):
+        g = networkx.Graph()
+        g.add_edge('a','b',weight=0.6)
+        g.add_edge('a','c',weight=0.2)
+        g.add_edge('c','d',weight=0.1)
+        g.add_edge('c','e',weight=0.7)
+        g.add_edge('c','f',weight=0.9)
+        g.add_edge('a','d',weight=0.3)
+        container = GraphContainer(graph=g)
+        for node in g.nodes():
+            GraphNodeComponent(container=container, value=node)
+        gc = GraphicsContext(tuple(container.bounds))
+
+        container.draw(gc)
+        self.assert_components_drawn(container)
+
+    def test_no_pygraphviz(self):
+        # monkey-patch pygraphviz_layout to raise ImportError
+        def fake_pygraphviz_layout(graph, prog):
+            raise ImportError
+
+        real_pygraphviz_layout = networkx.drawing.nx_agraph.pygraphviz_layout
```

Comment:
I would use
```
try: 
     ...
finally:
     networkx.drawing.nx_agraph.pygraphviz_layout = real_pygraphviz_layout
```
to make sure that networkx is not permanently changes when an error is fired

Even better, use the `mock` library to temporarily mock the function during the tests